### PR TITLE
Check duplicate tags

### PR DESCRIPTION
**Pull request by annstriganova**
Issue 5189.

### CODE REVIEW

File: src/main/java/com/axibase/tsd/collector/MessageHelper.java [hunk 1]
```diff
@@ -120,7 +121,10 @@ private void sendAggregatorEnvironmentProperty(WritableByteChannel writer) {
         for (Map.Entry<String, String> entry : environmentSettings.entrySet()) {
             String value = entry.getValue();
             if (!value.isEmpty()) {
-                sb.append(" v:").append(AtsdUtil.sanitizeName(entry.getKey())).append("=").append(AtsdUtil.sanitizeValue(value));
+                String tag = entry.getKey();
+                if (!StringUtils.containsIgnoreCase(sb, tag)) {
+                    sb.append(" v:").append(AtsdUtil.sanitizeName(tag)).append("=").append(AtsdUtil.sanitizeValue(value));
```
Comment (raipc) [inline src/main/java/com/axibase/tsd/collector/MessageHelper.java]: Please, store the filtered tags in a TreeMap. Extract it to a separate method (that gets a Map and returns a SortedMap) and   add unit tests. perform the same in `sendAggregatorRuntimeProperty` method (Properties implements Map).

Sergei, I think that removing keys with empty values decreases the usability of this collector. Let's remove check for emptiness.

Comment (raipc) [reply src/main/java/com/axibase/tsd/collector/MessageHelper.java]: It can be used as a flag, e.g.
`export ATSD_DEBUG`

**Comment by axibase-git** [comment]
At the moment property tags with empty values are discarded b ATSD and not stored.

On Apr 6, 2018, at 3:27 PM, Anton Rybochkin <notifications@github.com> wrote:

@raipc commented on this pull request.

In src/main/java/com/axibase/tsd/collector/MessageHelper.java <https://github.com/axibase/aggregation-log-filter/pull/6#discussion_r179742105>:

> @@ -120,7 +121,10 @@ private void sendAggregatorEnvironmentProperty(WritableByteChannel writer) {
         for (Map.Entry<String, String> entry : environmentSettings.entrySet()) {
             String value = entry.getValue();
             if (!value.isEmpty()) {
-                sb.append(" v:").append(AtsdUtil.sanitizeName(entry.getKey())).append("=").append(AtsdUtil.sanitizeValue(value));
+                String tag = entry.getKey();
+                if (!StringUtils.containsIgnoreCase(sb, tag)) {
+                    sb.append(" v:").append(AtsdUtil.sanitizeName(tag)).append("=").append(AtsdUtil.sanitizeValue(value));
It can be used as a flag, e.g.
export ATSD_DEBUG

**Comment by raipc** [comment]
Thanks, now I see. It's worth documenting in https://github.com/axibase/atsd/blob/master/api/network/property.md

**Comment by raipc** [comment]
We may need to do the same filtering for java properties

**Pull request merged**